src/averageAges.js

```diff
@@ -59,6 +83,28 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const ageDifference = [];
+
+  const peopleFiltered = people.filter((person) => onlyWithSon
+    ? person.sex === 'm' && person.mother !== null
+    : person.mother !== null);
+
+  const motherFiltered = people.filter((person) =>
+    people.some((mother) => mother.mother === person.name));
+
+  peopleFiltered.filter((person) =>
```

Comment:
you use a filter just to iterate the array (the same as the common `for` loop or `forEach` method) - change the logic to avoid it